### PR TITLE
Split up server/chat-commands/

### PR DESCRIPTION
**Pull request by Zarel**
`server/chat-commands.js` is now a directory. It's been split into `core`, `moderation`, and `admin`. `info` and `roomsettings` from `chat-plugins` have also moved to `chat-commands`.

Bot commands for inserting HTML into rooms have been moved from `info` into `admin`.

`/a` has been renamed `/addline`, for clarity (and also moved from `info` into `admin`).

### CODE REVIEW

**Comment by Zarel** [comment]
(I'll be honest, I did this mostly because I wanted chat-commands to be TypeScripted, but VS Code was lagging on typechecking it because it was too big; this will let the parts pe TypeScripted piecemeal.)

**Comment by panpawn** [comment]
Semi-unrelated, but I'm starting to wonder if ``server/`` should have a sub-directory called ``chat/``

File: server/chat-commands/core.js [hunk 1]
```diff
@@ -0,0 +1,1629 @@
...
+		if (targetUser.id in targetRoom.users) return this.errorReply(`This user is already in "${targetRoom.title}".`);
+
+		return `/invite ${targetRoom.roomid}`;
+	},
+	invitehelp: [
+		`/invite [username] - Invites the player [username] to join the room you sent the command to.`,
+		`(in a PM) /invite [roomname] - Invites the player you're PMing to join the room [roomname].`,
+	],
+
+	pminfobox(target, room, user, connection) {
```
Comment (whalemer) [inline server/chat-commands/core.js]: Should these be in admin with the rest of the html commands?

**Comment by scheibo** [comment]
> `/a` has been renamed `/addline`, for clarity (and also moved from info into admin).

~Hmm... seems like it could use an `/a` alias to continue to facilitate easy admin-abuse~

File: server/chat-commands/core.js (hunk 1 above)
Comment (Zarel) [reply server/chat-commands/core.js]: I actually do like them in html commands, actually.

**Comment by Zarel** [comment]
> ~Hmm... seems like it could use an `/a` alias to continue to facilitate easy admin-abuse~

In case you're half-serious, I think the up key is easy enough that I don't need to make it any easier.

**Pull request merged**